src/rustllvm/PassWrapper.cpp

```diff
@@ -1163,6 +1163,20 @@ extern "C" void LLVMRustSetModulePIELevel(LLVMModuleRef M) {
   unwrap(M)->setPIELevel(PIELevel::Level::Large);
 }
 
+//  Linking object files with different code models is undefined behavior
+//  because the compiler would have to generate additional code (to span
+//  longer jumps) if a larger code model is used with a smaller one.
+//  Therefore we will treat attempts to mix code models as an error.
```

Comment:
This last sentence is super interesting, but it is not super clear where the "error" comes from. Does it happen at LTO time? Quick skim of the linked differentials does not answer it, you need to read the comments, so it would be great if more about this was said here.

I would also probably move it to this function’s call location as people are way more likely to read it in `librustc_codegen_llvm` than here.

Reply:
> This last sentence is super interesting, but it is not super clear where the "error" comes from. Does it happen at LTO time? Quick skim of the linked differentials does not answer it, you need to read the comments, so it would be great if more about this was said here.

Thank you for having a look, and sorry for not clear that.
I checked LLVM and IRMover works both link time and LTO time, so this happends either.However, this error won't happen at rustc comiple time, so I would just remove this line.